test/test_nn.py

```diff
@@ -9067,6 +9067,17 @@ def test_batchnorm_buffer_update_when_stats_are_not_tracked(self):
         self.assertTrue(torch.equal(running_mean, bn.running_mean))
         self.assertTrue(torch.equal(running_var, bn.running_var))
 
+    @unittest.skipIf(not torch.cuda.is_available(), "CUDA not available")
+    def test_batchnorm_nhwc_cuda(self):
+        (N, C, H, W) = 2, 64, 50, 50
+        model = torch.nn.BatchNorm2d(C, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
+        model = model.eval().cuda().half()
+        inp1 = torch.randn(N, C, H, W, device=torch.device('cuda'), dtype=torch.half)
```

Comment:
While we're at it, do we want to span a wider net for dtypes, i.e. float32? Or is that already covered?

Reply:
That might depend on the intent of the test. If I understand correctly, float32 with channels last would dispatch to cudnn batchnorm rather than aten batchnorm.

Reply:
I'd say it could be worth it to generally cover channels last support for BatchNorm2d. If float32 isn't already covered by another test, then extending this one could prevent future regressions / errors. I'd assume that the fallback kernel should work whether cudnn is available or not, so it's not needed to guard against cudnn being available for this test.